projects/wave-core-new/src/lib/datasets/upload/upload.component.ts

```diff
@@ -0,0 +1,188 @@
+                                    "x": 180.0,
+                                    "y": 90.0
+                                }
+                            },
+                            "file_not_found_handling": "NoData"
+                        },
+                        "placeholder": "%%%_START_TIME_%%%",
+                        "time_format": "%Y-%m-%d",
+                        "start": 1388534400000,
+                        "step": {
+                            "granularity": "Months",
+                            "step": 1
+                        }
+                    }
+                }
+            }`,
+        },
+        {
+            name: 'Ports',
+            json: `{
+                "properties": {
+                    "id": null,
+                    "name": "Uploaded Natural Earth 10m Ports",
+                    "description": "Ports from Natural Earth",
+                    "source_operator": "OgrSource"
+                },
+                "meta_data": {
+                    "OgrMetaData": {
+                        "loading_info": {
+                            "file_name": "operators/test-data/vector/data/ne_10m_ports/ne_10m_ports.shp",
+                            "layer_name": "ne_10m_ports",
+                            "data_type": "MultiPoint",
+                            "time": "none",
+                            "columns": {
+                                "x": "",
+                                "y": null,
+                                "numeric": [
+                                    "natlscale"
+                                ],
+                                "decimal": [
+                                    "scalerank"
+                                ],
+                                "textual": [
+                                    "featurecla",
+                                    "name",
+                                    "website"
+                                ]
+                            },
+                            "default_geometry": null,
+                            "force_ogr_time_filter": false,
+                            "on_error": "skip",
+                            "provenance": null
+                        },
+                        "result_descriptor": {
+                            "data_type": "MultiPoint",
+                            "spatial_reference": "EPSG:4326",
+                            "columns": {
+                                "website": "Text",
+                                "name": "Text",
+                                "natlscale": "Number",
+                                "scalerank": "Decimal",
+                                "featurecla": "Text"
+                            }
+                        }
+                    }
+                }
+            }`,
+        },
+    ];
+
+    loadingInfo = '';
+
+    constructor(private dataSetService: DataSetService, private notificationService: NotificationService) {}
```

Comment:
if you set this to protected, we could write an app that extends this service and accesses the services.
So I try to use protected in our core library.

Reply:
7b114b945855735e9b45be3e93d131ce8ee7199c